go/adbc/README.md

```diff
@@ -0,0 +1,251 @@
+# adbc
+
+The adbc package provides:
+
+- Go Interfaces to [Arrow Database Connectivity](https://arrow.apache.org/adbc) (ADBC) for building ADBC drivers
+- An ADBC Driver Manager package (`drivermgr`) for loading and using ADBC Drivers
+- Scaffolding for writing new drivers in Go (`pkg`)
```

Comment:
the scaffolding is in `internal/driverbase`. `pkg` is for wrapping drivers for use outside of Go, generating the C shared lib